pyterrier/terrier/retriever.py

```diff
@@ -13,7 +14,30 @@
 from pyterrier.terrier.stopwords import TerrierStopwords
 from pyterrier.terrier.tokeniser import TerrierTokeniser
 
+def _query_needs_tokenised(query: str) -> bool:
```

Comment:
This feels a bit brittle, but maybe okay?

The way I did it in pyterrier-anserini is to have multiple fields for the query. `query` representing the natural-langauge query, and `query_lucene` for the lucene query langauge version of the query. (This mimics how we handle the bow `query_toks` type of query, too).

This would require changes throughout several components, but I guess this is the type of breaking change we can do in a 1.0 release.

Reply:
i know you are a proponent of this approach - I don't feel ready for this yet.